model.go

```diff
@@ -8,9 +8,10 @@ import (
 
 // Model is the base type of the items that are stored
 type Model struct {
-	ID        ID
-	persisted bool
-	writable  bool
+	ID             ID
+	virtualColumns []VirtualColumn
```

Comment:
maybe not, i have to fix some stuff in there, so I will try to see if it can be private